src/gui/optionsdialog.cpp

```diff
@@ -621,38 +630,15 @@ void OptionsDialog::changePage(QListWidgetItem *current, QListWidgetItem *previo
     m_ui->tabOption->setCurrentIndex(m_ui->tabSelection->row(current));
 }
 
-void OptionsDialog::loadWindowState()
-{
-    Utils::Gui::resize(this, Preferences::instance()->getPrefSize());
-}
-
 void OptionsDialog::loadSplitterState()
 {
-    const QStringList sizesStr = Preferences::instance()->getPrefHSplitterSizes();
-
     // width has been modified, use height as width reference instead
     const int width = Utils::Gui::scaledSize(this
         , (m_ui->tabSelection->item(TAB_UI)->sizeHint().height() * 2));
-    QList<int> sizes {width, (m_ui->hsplitter->width() - width)};
-    if (sizesStr.size() == 2)
-        sizes = {sizesStr.first().toInt(), sizesStr.last().toInt()};
-    m_ui->hsplitter->setSizes(sizes);
-}
-
-void OptionsDialog::saveWindowState() const
-{
-    Preferences *const pref = Preferences::instance();
-
-    // window size
-    pref->setPrefSize(size());
-
-    // Splitter size
-    const QStringList sizesStr =
-    {
-        QString::number(m_ui->hsplitter->sizes().first()),
-        QString::number(m_ui->hsplitter->sizes().last())
-    };
-    pref->setPrefHSplitterSizes(sizesStr);
+    const QStringList strings = m_storeHSplitterSize.get({QString::number(width)
+        , QString::number(m_ui->hsplitter->width() - width)});
+    Q_ASSERT(strings.size() == 2);
```

Comment:
I wouldn't assert on any kind of external data. It's better to use some defaults in case of incorrect input.

Reply:
>It's better to use some defaults in case of incorrect input.

We already provide default values but it is hard to be sure it is correct.

>I wouldn't assert on any kind of external data.

Maybe a assertion is really too much... removed.

Reply:
>We already provide default values but it is hard to be sure it is correct.

These values are used only if there is no corresponding setting in the configuration file. But I meant the case when there is incorrect data. It can be left to chance in some simple cases. But in this case, it can lead, for example, to an access outside the bounds of the array. So you have to check that the resulting list corresponds to the expected size, and adjust it in case of a mismatch.

Reply:
>It can be left to chance in some simple cases. But in this case, it can lead, for example, to an access outside the bounds of the array.

I took care of it now.

>So you have to check that the resulting list corresponds to the expected size, and adjust it in case of a mismatch.

Now we only provide sane default sizes. If the user modifies the setting and broke it (less/more size parameters), then let it be.